### ISSUE TITLE
Swarms of size 20,50,100 don't complete

### ISSUE THREAD

**Issue by jackpal** [jackpal/Taipei-Torrent#55]
Repro: Uncomment the commented-out-tests in tracker_test.go

Might be some sort of OS-level resource starvation or race condition. Tested on OSX. Fails reproducibly. (Which tends to indicate that it's not a race condition.)

**Comment by rkjdid** [comment]
I tried to debug this issue, 2 different bugs seem to produce a fail on a debian 8

1st thing is to make sure that all the ports used for testing the swarm are available, right now it is hardcoded from 7000 to 7000+1+n (n being the size of the swarm, 7000 is seed). If a port is already in use the client just fails to run. a `grep "Listen failed"` on the output shows the "failing" nodes. I submitted PR #100 with a small fix regarding the use of port 0, to avoid this case of unavailable ports

once this is put aside, the test randomly pass or fails with timeout, with some log.Print I found that the timeouts occur because 1 or more clients don't properly exit, [the `p.cmd.Wait()` blocks](https://github.com/jackpal/Taipei-Torrent/blob/master/tracker/tracker_test.go#L84) while the loop popping the `doneCh` is waiting for these last few to return (rarely more than 1). I didn't find where they are stuck, but a diff on the data file for the corresponding hanged clients show that it was properly downloaded. Also they don't seem to be in the `DoTorrents` loop since at this point in the test I only see the heartbeat log of the `seed`, the missing leech clients don't appear anymore.
